### PR TITLE
Apply out of the box modern theme

### PR DESCRIPTION
**Pull request by anoopt**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| New sample?      | no 
| Related issues?  | NA

#### New bits
This PR has a new extension method called "ApplyOOBModernTheme" which can be used to apply an out of the box modern theme to a site. 
The JSON for theme is created by a new method called "GetOOBModernThemeJson" in "ThemeUtility.cs". 
There is also an enum called "OOBTheme" which has all the out of the box theme names.

#### Existing bits
Thanks to Paolo, the already existing "ApplyThemeAsync" is used for calling the "ApplyTheme" endpoint. 

Thanks to Gautam Sheth for the PR suggestion.

### CODE REVIEW

File: Core/OfficeDevPnP.Core/Utilities/Themes/ThemeUtility.cs [hunk 1]
```diff
@@ -48,6 +48,36 @@ public static String GetThemeAsJSON(String primaryColor, String bodyTextColor, S
...
         }
+
+        /// <summary>
+        /// Return an out of the box modern theme as a JSON string
+        /// </summary>
+        /// <param name="theme">Name of the out of the box theme</param>
+        /// <returns></returns>
+        public static String GetOOBModernThemeJson(OOBTheme theme)
+        {
+            switch (theme)
```
Comment (heinrich-ulbricht) [inline Core/OfficeDevPnP.Core/Utilities/Themes/ThemeUtility.cs]: Interesting. I just had a look at the "Change the Look" options and the theme colors. They seem to be embedded in the JS bundle. So no API to get the colors from (which I expected). But how do we detect if the default colors change? (Does it even matter?)

Comment (anoopt) [reply Core/OfficeDevPnP.Core/Utilities/Themes/ThemeUtility.cs]: True. Even I did not see any API which provides the OOB theme information. So one option is to use the code in the PR and as soon as there is one I will update the code.

**Comment by erwinvanhunen** [comment]
Hi! Thank you for the PR. However, we merged another PR (#2121) which does exactly the same, in a somewhat different (but similar) implementation. Unfortunately this means we will not merge this PR. Nevertheless, thank you very much for your efforts! We really appreciate your work.

**Comment by anoopt** [comment]
Ah. I didn't know there was another PR for this. My bad. Apologies.

**Pull request closed without merging**